src/main/scala/org/dia/algorithms/mcs/MCCOps.scala

```diff
@@ -0,0 +1,237 @@
+        if (isPotentialMCCNode) {
+          for (path <- MCCNodeSet) {
+            if (path.size > 0) {
+              val tempSet = new mutable.HashSet[String]()
+              tempSet ++= path
+              tempSet += node
+              if (globalMCCs.contains(node)) {
+                globalMCCs(node) += tempSet
+              }
+              else {
+                val tempList = new mutable.MutableList[mutable.HashSet[String]]()
+                tempList += tempSet
+                globalMCCs(node) = tempList
+              }
+            }
+          }
+        }
+      }
+    }
+    if (isPotentialMCCNode) {
+      if (!globalMCCs.contains(node)) {
+        val path = new mutable.HashSet[String]()
+        path += node
+        val tempList = new mutable.MutableList[mutable.HashSet[String]]()
+        tempList += path
+        globalMCCs(node) = tempList
+      }
+      return globalMCCs(node)
+    }
+    else return new mutable.MutableList[mutable.HashSet[String]]()
+  }
+
+  /**
+   * Check if a given node satisfies the two criterions specified by Laurent et al 1998
+   * @param node The MCS Node to check
+   * @param areaCriteriaA
+   * @param areaCriteriaB
+   * @param tempCriteriaA
+   * @param tempCriteriaB
+   * @return A tuple with a boolean indicating if the node satisfies the criteria or not
+   */
+  def isPotentialNode(node : MCSNode,
+      areaCriteriaA: Int,
+      areaCriteriaB: Int,
+      tempCriteriaA: Float,
+      tempCriteriaB: Float):
+      (Boolean, String) = {
+
+    /** Area within the node matching the temp criteria for A acc to Laurent et al 1998 */
+    var nodeAreaCriteriaA: Int = 0
+    /** Area within the node matching the temp criteria for B acc Laurent et al 1998 */
+    var nodeAreaCriteriaB: Int = 0
+
+    for ((pos, temp) <- node.grid) {
+      // Acc to Laurent et al temp for criteria ! is less than criteria B
+      if (temp <= tempCriteriaA) {
+        nodeAreaCriteriaA += 1
+        nodeAreaCriteriaB += 1
+      }
+      else if (temp <= tempCriteriaB) {
+        nodeAreaCriteriaB += 1
+      }
+    }
+    if (nodeAreaCriteriaA >= areaCriteriaA || nodeAreaCriteriaB >= areaCriteriaB) {
+      return (true, node.hashKey())
+    }
+    else {
+      return (false, node.hashKey())
+    }
+  }
+
+  /**
+   * Load the subgraphs found in the previous steps from the files spcified in filepath
+   * @param filepath
+   * @param sc
+   * @return
+   */
+  def loadSubgraphsFromFile(filepath: String, sc: SparkContext):
+  RDD[mutable.MutableList[(String, String)]] = {
+    val textfiles = sc.textFile(filepath).map(x => {
+      val edgesString = x.replace("Set(", "")
+        .replace(")", "")
+        .split(", ")
+      val edgesTupleList = new mutable.MutableList[(String, String)]
+      for (edge <- edgesString) {
+        val nodes = edge.split(",")
+        edgesTupleList += ((nodes(0), nodes(1)))
+      }
+      edgesTupleList
+    })
+    return textfiles
+  }
```

Comment:
combining sets with overlapping nodes to make one set is required.
For e.g. running with another dataset, the following was output:
Set(2009083120:18, 2009083117:16, 2009083123:26, 2009083118:11, 2009083119:16, 2009083122:22, 2009083121:9)Set(2009083117:16, 2009083112:15, 2009083116:19, 2009083113:22, 2009083115:14, 2009083114:16)

The node 2009083117:16 repeats in both sets, thus linking the two, and the MCC should really be the combination of these two.